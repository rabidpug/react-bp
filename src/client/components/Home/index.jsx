import React from 'react';
import gStyles from 'Styles/global';

const Home = () => (
  <div className={ gStyles.cardStyle }>
    <div className={ gStyles.cardHeader }>
      <h1>Welcome to React ENFRAMR!</h1>
    </div>
    <div className={ gStyles.cardTop }>
      <p> A Full Stack React Boilerplate with lots of stuff built in to save me time.</p>
    </div>
  </div>
);

export default Home;
