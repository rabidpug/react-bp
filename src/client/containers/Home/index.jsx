import { subscribePush, unsubscribePush, } from 'Client/webpush';

import Button from 'Components/Button';
import Card from 'Components/Card';
import React from 'react';

const { Container, Header, Body, Footer, } = Card;
const Home = () => (
  <Container>
    <Header>
      <h1>Welcome to React ENFRAMR!</h1>
    </Header>
    <Body>
      <p> A Full Stack React Boilerplate with lots of stuff built in to save me time.</p>
    </Body>
    <Footer>
      <Button
        onClick={ subscribePush }
        variant='primary'>
        Subscribe
      </Button>
      <Button
        onClick={ unsubscribePush }
        variant='secondary'>
        Unsubscribe
      </Button>
    </Footer>
  </Container>
);

export default Home;
