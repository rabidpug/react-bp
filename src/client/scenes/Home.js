import LoginUserPass from 'Containers/LoginUserPass';
import React from 'react';

const Home = () => (
  <div>
    Welcome to React ENFRAMR!<br />
    <LoginUserPass />
  </div>
);

export default Home;
