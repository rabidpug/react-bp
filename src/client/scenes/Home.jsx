import FadeWrap from 'Animations/FadeWrap';
import React from 'react';

const Home = FadeWrap( ( { style, } ) => <div style={ { ...style, } }>Welcome to React ENFRAMR!</div> );

export default Home;
