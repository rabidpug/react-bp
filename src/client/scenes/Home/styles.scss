.bodyStyle {
  background-color: green;
  width: 90%;
  margin: auto;
  display: block;
}
