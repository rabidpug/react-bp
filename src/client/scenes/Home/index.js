import HelloButton from '../../containers/HelloButton';
import Menu from '../../containers/Menu';
import Message from '../../containers/Message';
import React from 'react';
import Routes from '../../containers/Routes';
import styles from './styles.scss';

const Home = () => (
  <div className={ styles.body }>
    <Menu />
    <Message />
    <HelloButton />
    <Routes />
  </div>
);

export default Home;
