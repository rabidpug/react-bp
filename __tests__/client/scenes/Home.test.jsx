import Home from 'Scenes/Home';
import React from 'react';
import { shallow, } from 'enzyme';
import { shallowToJson, } from 'enzyme-to-json';

describe(
  'TopBar', () => {
    it(
      'Should render the Home page', () => {
        const wrapper = shallow( <Home /> );

        expect( shallowToJson( wrapper ) ).toMatchSnapshot();
      }
    );
  }
);
