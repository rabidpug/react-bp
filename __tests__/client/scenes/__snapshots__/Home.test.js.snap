// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar Should render the Home page 1`] = `
<div>
  Welcome to React ENFRAMR!
</div>
`;
